### DISCUSSION TITLE
Why different inputs `text_l` and `text_g` in CLIPTextEncodeSDXL?

### DISCUSSION THREAD

**Discussion by ifsheldon** [comfyanonymous/ComfyUI#1047] [Q&A]
I notice that there are two inputs `text_g` and `text_l` to `CLIPTextEncodeSDXL `. I skimmed through the [SDXL technical report](https://arxiv.org/abs/2307.01952) and I think these two are for OpenCLIP ViT-bigG and CLIP ViT-L. 

I don't know why Stability wants two CLIPs, but I think the input to the two CLIPs can be the same. So, @comfyanonymous perhaps can you tell us the motivation of allowing the two CLIPs to have different inputs? Did you find interesting usage?

https://github.com/comfyanonymous/ComfyUI/blame/4a77fcd6ab01d69e18c384faa29ae1c3d02237f3/comfy_extras/nodes_clip_sdxl.py#L41C5-L41C5

Thanks!

**Comment by yuhuihu** [comment]
I also have the same question.  Here's som info I founded: https://www.youtube.com/watch?v=zucLSuCvqM8

**Comment by okolenmi** [comment]
I tested it on my own. It looks like G is good in details/concept and L is good in image content.

L is like "Charcter with long green hair on white background"
G is like "Angry face expression, crossed hands"

G is bad in things like character traits (age, clothes, etc.) L is good in this things. G is better than L in complicated concept things. You can add characters in L and then change what they doing in G.

**Comment by mcandre** [comment]
In practice, lots of users are manually copying and pasting the same prompt into both parameters.

Computers exist to automate. Let's create an option to automatically sync these fields.

**Reply by asagi4** [reply]
@mcandre if you just use the standard CLIPTextEncode node instead of the SDXL-specific one, it will use the same text for both TEs. Or you can just use a primitive text node and connect it to both inputs.
